avocado/utils/software_manager.py

```diff
@@ -57,7 +51,14 @@
 log = logging.getLogger('avocado.test')
 
 
-SUPPORTED_PACKAGE_MANAGERS = ['apt-get', 'yum', 'zypper', 'dnf']
+def _get_backends():
+    """
+    Get the mapping of package manager name to implementation class.
+
+    Note: New implementations should have an entry added in the map.
+    """
+    return {'apt-get': AptBackend, 'yum': YumBackend, 'dnf': DnfBackend,
+            'zypper': ZypperBackend}
```

Comment:
Why this, instead of the previous global var? Any specific reason? Just trying to understand.

Reply:
Because the keys (apt-get, yum, etc) were repeated in the SUPPORTED_PACKAGE_MANAGER (list) and in the embedded dict on line 144.

Reply:
@clebergnu @wainersm ... 

Sorry if I was not clear. But I mean: why not the dict as global var?

```python
PACKAGE_MANAGER_BACKENDS = {'apt-get': AptBackend,
                            'yum': YumBackend,
                            'dnf': DnfBackend,
                            'zypper': ZypperBackend}
```

Reply:
I agree... that's what I proposed here: https://github.com/avocado-framework/avocado/pull/3571#discussion_r381543877

Reply:
So I guess @beraldoleal is the third person supporting this ^^

Reply:
Yes, I will send a v3 where it is a global variable. I choose to use a method because I wanted it declared at beginning of the module, with documentation clearly stating that new implementers should extend the map.

Reply:
@wainersm please keep the documentation, using `#:` on top of the variable definition so that it gets published to our API reference.